src/ERC4626Migrator.sol

```diff
@@ -5,28 +5,33 @@ import {ERC20} from "solmate/tokens/ERC20.sol";
- * Practically LP tokens that can be migrated to ETH, DAI, and USDC.
- * Eth, Dai and USDC held by the contract will be used to distribute to users, so that the contract
+ * Practically LP tokens that can be migrated to WETH, DAI, and USDC.
+ * WETH, Dai and USDC held by the contract will be used to distribute to users, so that the contract
  * is funded before users start using it, as they otherwise could simply sacrifice their share of the
  * assets.
  */
 contract ERC4626Migrator is ReentrancyGuard {
     using FixedPointMathLib for uint256;
     using SafeTransferLib for ERC20;
 
-    error InvalidSignature(address user, bytes signature);
+    error InvalidAcceptanceToken(address user, bytes32 acceptanceToken);
 
-    event Migrated(address indexed user, uint256 amount, uint256 ethAmount, uint256 daiAmount, uint256 usdcAmount);
+    event MigratedAndAgreed(address indexed user, uint256 amount, uint256 wethAmount, uint256 daiAmount, uint256 usdcAmount);
 
-    // @todo Replace this with the real hash
-    bytes32 public constant TOS = bytes32("TOS");
+    /**********
+    By clicking "I agree to the terms to claim redemption" on the euler.finance web interface or executing the EulerClaims smart contract and accepting the redemption, I hereby irrevocably and unconditionally release all claims I (or my company or other separate legal entity) may have against Euler Labs, Ltd., the Euler Foundation, the Euler Decentralized Autonomous Organization, members of the Euler Decentralized Autonomous Organization, and any of their agents, affiliates, officers, employees, or principals related to this matter, whether such claims are known or unknown at this time and regardless of how such claims arise and the laws governing such claims (which shall include but not be limited to any claims arising out of Euler’s terms of use).  This release constitutes an express and voluntary binding waiver and relinquishment to the fullest extent permitted by law.  If I am acting for or on behalf of a company (or other such separate entity), by clicking "I agree to the terms to claim redemption" on the euler.finance web interface or executing the EulerClaims smart contract and accepting the redemption and agreement, I confirm that I am duly authorised to enter into this contract on its behalf.
 
+    This agreement and all disputes relating to or arising under this agreement (including the interpretation, validity or enforcement thereof) will be governed by and subject to the laws of England and Wales and the courts of London, England shall have exclusive jurisdiction to determine any such dispute.  To the extent that the terms of this release are inconsistent with any previous agreement and/or Euler’s terms of use, I accept that these terms take priority and, where necessary, replace the previous terms.
+    **********/
+    
+    bytes32 public constant TOS = 0xc363fe4cdd3f22305ac245691fcdaee3ea8031ac1d5078fb042f886b7aec89c7;
```

Comment:
Please add a not on how this is computed. Just using the message directly here did not seem to give me the correct hash.